### ISSUE TITLE
Cannot Connect to wifi

### ISSUE THREAD

**Issue by vitalikkh** [Blackymas/NSPanel_HA_Blueprint#1493]
Hi,

And Happy New Year!

For a couple days I am desperately trying to install ESP Home on the original US version of NSPanel (not Pro).
Everything looks nice and clean. Validation did not show any errors. Also project compiles clean.
Here is a Configuration File I am using:
```
substitutions:
  ###### CHANGE ME START ######
  device_name: "nspanel" 
  wifi_ssid: !secret wifi_ssid
  wifi_password: !secret wifi_password

  nextion_update_url: "http://homeassistant.local:8123/local/nspanel_us.tft"
  nextion_blank_url: "http://homeassistant.local:8123/local/nspanel_blank.tft"

  ##### addon-configuration #####
  ## addon_climate ##
  # addon_climate_heater_relay: "1" # possible values: 1/2

  ##### CHANGE ME END #####

packages:
  remote_package:
    url: https://github.com/Blackymas/NSPanel_HA_Blueprint
    ref: main
    files:
      - nspanel_esphome.yaml # Core package
      # - advanced/esphome/nspanel_esphome_advanced.yaml # activate advanced (legacy) elements - can be useful for troubleshooting
      # - nspanel_esphome_addon_climate_cool.yaml # activate for local climate (cooling) control
      # - nspanel_esphome_addon_climate_heat.yaml # activate for local climate (heater) control
    refresh: 1s

esp32:
  framework:
    type: esp-idf

##### My customization - Start #####
##### My customization - End #####

```
Flashing went without any issues. But obviously some thing is not right.
The unexpected device "ESPHome Web 549ee4" appeared under ESP Home tab: 
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/105172911/600c79a4-3f65-4865-86e3-29af832827f2)

This one is actually on line and it is definitely physical NSPanel device.
Here is what I can see when connection to this device:
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/105172911/4a70f77d-6659-4c52-b3bd-66679b38696e)

However my device "nspanel" is unresponsive and produces next log on the reboot:

INFO ESPHome 2023.12.5
INFO Reading configuration /config/esphome/nspanel.yaml...
INFO Updating https://github.com/Blackymas/NSPanel_HA_Blueprint@main
INFO Detected timezone 'America/New_York'
INFO Starting log output from nspanel.local using esphome API
WARNING Can't connect to ESPHome API for nspanel.local: Error resolving IP address: [Errno -5] No address associated with hostname (APIConnectionError)
INFO Trying to connect to nspanel.local in the background

What I am doing wrong or what steps I am missing?

Thank you in advance,

- Vitaliy

PS.
Basically I heed very simple Scene Controller with 6 to 8 user configurable Buttons. I need to assign Icon and Description Text
to each Button individually. And of course, I am planning to control 2 built-in relays by command (not from panel physical buttons).

**Comment by edwardtfn** [comment]
On the ESPHome Web installer, you probably clicked on "Prepare for first use" instead of "Install". That is not a problem, ax you have access to its Web interface and can upload the firmware from there without the need of connecting wires again.

On the ESPHome dashboard, click on Install and select the option for "Manual download". That will provide you a `.bin` file on your download folder. With that file, go back to the device's Web interface (your last screenshot), click the button to Upload OTA firmware,point to the `.bin` file and follow the instructions... It shouldn't take more than a few seconds and your panel will restart with the new firmware loaded.

**Comment by vitalikkh** [comment]
First of all - Thank you very much for the very quick response!
You are absolutely correct. I did click on "Prepare for first use" because "Install" for some reason was not clickable at that time.
I followed yours instruction and Panel was up and running. But after recircling power it reverted back  and that ESPHome Web
devices popped up again. I repeated this process twice and result is the same. To my eyes it looks like the *.bin file is
loaded directly into RAM instead of being written into flash.
What I am missing now?

**Comment by vitalikkh** [comment]
Update.
I reflash panel first from that ESPHome Web device and while panel was running I reflash it again from its interface wirelessly.
Now I can recycle power and panel is starting up correctly.
Next step will be to configure everything according to my application.
Thank you very much for your help.

Happy New Year!,

- Vitaliy

**Comment by vitalikkh** [comment]
Actually I have a problem with updating TFT Display.
The related screen is a bit different from the video tutorial. I cannot see that button which starts the TFT update process:
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/105172911/42ab35b0-f9df-4f6e-94e0-671fc3d31210)

Please advice where is this button or what is another way to update TFT display?

Thank you,

- Vitaliy
-

**Comment by edwardtfn** [comment]
> The related screen is a bit different from the video tutorial. I cannot see that button which starts the TFT update process:
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/105172911/42ab35b0-f9df-4f6e-94e0-671fc3d31210)

This is the right button... What happens when you press this one? Could you please share the logs from ESPHome?

**Comment by edwardtfn** [comment]
> And actually Diagnostic shows the Display is disconnected:

> image

> I guess this is why there is no button for the display updating.
Please advice how to fix the issues with the display?

This will show "Disconnected" until the right TFT is uploaded.

**Comment by vitalikkh** [comment]
Log file is attached:
[logs_nspanel_logs.txt](https://github.com/Blackymas/NSPanel_HA_Blueprint/files/13801101/logs_nspanel_logs.txt)

It looks like something is not right with file:  [nspanel_us.tft](https://github.com/Blackymas/NSPanel_HA_Blueprint/raw/main/nspanel_us.tft)

**Comment by edwardtfn** [comment]
> It looks like something is not right with file: [nspanel_us.tft](https://github.com/Blackymas/NSPanel_HA_Blueprint/raw/main/nspanel_us.tft)

That's exactly the issue. Your file is too small.
Please try to download the file from GitHub again and upload it to your local server again.
You can always use your browser to download from your local server (http://homeassistant.local:8123/local/nspanel_us.tft) and double check the file size if it is the same as the one from GitHub.

**Comment by edwardtfn** [comment]
https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-Common-Issues-TFT-Upload#check-the-file-size-by-downloading-to-your-computer

**Comment by vitalikkh** [comment]
File is not accessible. I am getting this error message:
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/105172911/38c46ce7-0a4f-4c27-a796-8550c408834d)

What is a correct file location on the HA.
I created www folder under config folder (\\192.168.20.93\config\www)
Is this a correct location for local files?

**Comment by edwardtfn** [comment]
Yest, it should be www folder under config. Can you see that with a file editor in your home assistant? Is the file there?

**Comment by edwardtfn** [comment]
Sorry, looks like that config folder was renamed to homeassistant recently, so it shoukd be under homeassistant/www

**Comment by vitalikkh** [comment]
This what I am seeing:
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/105172911/0ee104ca-6289-4120-a551-0a05644cc2c6)

It looks like all tft files are in the  homeassistant/www folder.

But for some reason the URL is not accessible:
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/105172911/d26d084f-a254-49ee-b7b4-268940dcb723)

**Comment by edwardtfn** [comment]
Have you tried restarting Home Assistant?

**Comment by vitalikkh** [comment]
Yes, I did restart many times HA and Panel.
So far no matter what I am trying I cannot access tft files from any URL I am/was trying.
However I can access HA from Windows File Explorer and from HA File editor.

**Comment by vitalikkh** [comment]
OK, I am running two Hubitat Elevation Hubs (C7 and C8) which are capable to store user files and these stored files are accessible via HTTP URL. I re-flashed NSPanel with modified URLs for the accessing TFT related files. This time the TFT update process was able to access and validate TFT files but still failed. The related log file is attached:
[logs_nspanel_run_02.txt](https://github.com/Blackymas/NSPanel_HA_Blueprint/files/13801661/logs_nspanel_run_02.txt)

So, what is wrong now?

**Comment by edwardtfn** [comment]
Look, the problem with not starting the update is related to your the display (Nextion) not responding the transfer request from ESPHome. This could be related to baud rate or a cable disconnected, but I see at your logon dump info that the connection to the panel was established at 115200bps:

```log
[11:03:05][C][nextion:128]: Nextion:
[11:03:05][C][nextion:129]:   Device Model:     NX4832F035_011C
[11:03:05][C][nextion:130]:   Firmware Version: 51
[11:03:05][C][nextion:131]:   Serial Number:    6AB134019EB1AD3E
[11:03:05][C][nextion:132]:   Flash Size:       16777216
[11:03:05][C][nextion:133]:   Wake On Touch:    YES
[11:03:05][C][nextion:134]:   Exit reparse:     NO
[11:03:05][C][nextion:145]:   Start Up Page:    8
```

So, it isn't neither baud rate or cable, but due to an incorrect command:

```log
[11:05:22][D][script.upload_tft:144]: Wait 5 seconds
[11:05:27][D][script.upload_tft.upload_tft_esp_idf:722]: TFT file size: 4294967295
[11:05:27][D][script.upload_tft.upload_tft_esp_idf:729]: File size check passed. Proceeding...
[11:05:27][D][script.upload_tft.upload_tft_esp_idf:734]: Updating Nextion
[11:05:27][D][script.upload_tft.send_nextion_command:206]: Sending: whmi-wris -1,921600,1
[11:05:27][D][script.upload_tft.upload_tft_esp_idf:758]: Waiting for upgrade response
[11:05:29][D][script.upload_tft.upload_tft_esp_idf:763]: Upgrade response is []
[11:05:29][E][script.upload_tft.upload_tft_esp_idf:768]: Preparation for tft update failed 0 ""
[11:05:29][D][script.upload_tft.upload_end_:274]: Completed: 0 
```
The TFT size is showing 4Gb, and this is not correct. This is causing ESPHome to send an incorrect size (-1 byte) when starting the transfer and this is probably the reason it fails.

Could you please check the url with your browser again?

**Comment by vitalikkh** [comment]
OK, I used a URL from YAML file for file verification: nextion_update_url: "http://192.168.20.90/local/nspanel_us.tft"

Both files seem to be the same size (original file is nspanel_us.tft and what I downloaded back is nspanel_us_read_back.tft):
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/105172911/94e07424-cd87-42c0-88ac-8b764056911c)

Furthermore, I checked both files with Notepad++ and they are reported identical.

**Comment by edwardtfn** [comment]
Try this one: http://nspanel.costafirmo.com/tft/v4.1.4/nspanel_us.tft

**Comment by vitalikkh** [comment]
OK. Finally the TFT update is started and about 30% done.
I hope it will finish without problems.
I will let you know after its done ...

**Comment by vitalikkh** [comment]
OK, It finished I hope, successfully.
Display is connected and seems to be functional. It reported a correct Time and internal temp sensor value.
However log (file is attached) still reports:
[13:08:36][W][nextion:342]: Nextion reported invalid instruction!
[13:08:36][W][nextion:342]: Nextion reported invalid instruction!

as soon as i am touching screen.
I hope this is something benign.

Now I need to create a very simplified BLUEPRINT. Everything what I need - is a single page with 6 to 8 configurable
Buttons and of course, both relays.
Is it OK to delete everything what I don't need from the original BLUEPRINT?

And finally please email me how I can reward you for all your very fast and very valuable help?

Happy New Year!,

- Vitaliy

**Comment by edwardtfn** [comment]
You shouldn't be seen those error messages on your log. It isn't harmful and you can live with that if everything is working fine, but it's definitely a signal of something not 100% correct.
How often you see those messages? Can you see any pattern (a page being opened, a time interval, going to sleep, etc)?
When you restart your device, which version numbers you see for TFT, ESPHome and Blueprint? A version mismatch could cause this kind of message.

**Comment by edwardtfn** [comment]
> Now I need to create a very simplified BLUEPRINT. Everything what I need - is a single page with 6 to 8 configurable
Buttons and of course, both relays.
> Is it OK to delete everything what I don't need from the original BLUEPRINT?

You can set all the buttons in a single button page (max 8 buttons per page) and then go to device's page (Settings > Devices and Services > ESPHome) and set the following;
- Wake up page: choose the button page you set, so your panel will always wake-up at that page.
- Page timeout: set to 0 to prevent falling back to the home page (the one with the clock)

Please let me know how that fits your needs.
We are planing to add the possibility to hide other empty pages, but that will come later (currently planned for v4.3).

**Comment by vitalikkh** [comment]
All component (TFT, ESPHome and Blueprint) versions are 4.1.4
Pair of these messages comes only when screen reverts to Home Page.
[15:49:38][D][nextion_textsensor:014]: Processed text_sensor "currentpage" state "home"
[15:49:38][W][nextion:342]: Nextion reported invalid instruction!
[15:49:38][W][nextion:342]: Nextion reported invalid instruction!

OK, I will follow yours instruction for configuring the existing unmodified Blueprint.
I hope, this will be OK for my very minimal needs.

**Comment by edwardtfn** [comment]
Ok, thanks. I will try to reproduce these messages with a fresh new install.

I will close this issue here, as the original issue was solved. Please feel free to create another one if you face anything wrong or if you have any question.

And enjoy the new year. 😃
